apps/checkout/src/hooks/useSetFormErrors.ts

```diff
@@ -0,0 +1,37 @@
+import { FieldError, Path, UseFormSetError } from "react-hook-form";
+import { Errors } from "@/checkout/hooks/useErrors";
+import { forEach } from "lodash-es";
+import { useEffect } from "react";
+
+function useSetFormErrors<TFormData>(data: {
+  setError: UseFormSetError<TFormData>;
+  hasErrors?: boolean;
+  errors?: Errors<TFormData>;
+}): () => void;
+
+function useSetFormErrors<TFormData>(data: {
+  setError: UseFormSetError<TFormData>;
+}): (hasErrors?: boolean, errors?: Errors<TFormData>) => void;
+
+function useSetFormErrors<TFormData>({ setError, hasErrors, errors }: any) {
```

Comment:
I think we could type it a bit better and lose the `any`

Reply:
Same as above, the typing is done above and is intentional, either one of the types above will override the any